---
title: "Craniosynostosis RNAseq- Male Model Differential Gene Expression"
author: "Samantha Lapehn Young"
date: "4/21/2022"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Male Craniosynostosis RNAseq: Male Model Differential Gene Expression 
This code performs the differential expression analysis for the Cunningham Lab Craniosynostosis RNAseq data Male Samples.

## Load Packages
```{r Packages}
library(tidyverse)
library(limma)
library(RColorBrewer)
library(UpSetR)
library(org.Hs.eg.db)
library(edgeR)
library(pheatmap)
library(ChIPpeakAnno)
```


## Load Data
Loading data after surrogate variable analysis and voom transformation
```{r Load Data}
load("/Volumes/paquette_a/slapehn/CranioRNAseq/Male_CranioRNAseq/Data/Male_CranioRNAseq_VoomAdjustedData.RData")
```

## Save Expression Data for Supp Table
saving data for formatting a supplementary table 
```{r Expr Data}
genes <- v$genes
logcpm_male <- v$E
logcpm_male_DF <-data.frame(logcpm_male)
logcpm_male_DF$mean_logcpm <- rowMeans(logcpm_male_DF)
logcpm_male_DF$ensembl <- rownames(logcpm_male_DF)

logcpm_male_all <- full_join(logcpm_male_DF, genes, by=c("ensembl"="ensembl_gene_id")) %>%
  dplyr::select(hgnc_symbol, mean_logcpm)
write.csv(logcpm_male_all, "/Volumes/paquette_a/slapehn/CranioRNAseq/DEGs/logcpm_male.csv")
```

## Run Full Model on voom transformed data
Run differential gene expression model for each phenotype using the limma pipeline
```{r Differential expression}
#Step 1: Run model
fit <- eBayes(lmFit(v, mod))

#Step 2: Pull all results
AllResults_Coronal<-topTable(fit, coef=2,number=dim(fit)[1],adjust.method="BH") 
AllResults_Lambdoid <-topTable(fit, coef=3,number=dim(fit)[1],adjust.method="BH") 
AllResults_Metopic<-topTable(fit, coef=4,number=dim(fit)[1],adjust.method="BH") 
AllResults_Sagittal <-topTable(fit, coef=5,number=dim(fit)[1],adjust.method="BH") 

#Step 3: Pull only significant results (p.adj<0.05)
Sig_Coronal <- AllResults_Coronal %>%
  dplyr::filter(adj.P.Val<0.05)
Sig_Lambdoid <- AllResults_Lambdoid %>%
  dplyr::filter(adj.P.Val<0.05)
Sig_Metopic <- AllResults_Metopic %>%
  dplyr::filter(adj.P.Val<0.05)
Sig_Sagittal <- AllResults_Sagittal %>%
  dplyr::filter(adj.P.Val<0.05)
```

##DEG Volcano Plots
Visualize DEG significance as a volcano plot
```{r Volcano}
MakeVPlot<-function(Results,Title){
sig<-subset(Results,adj.P.Val<0.05)
sig<-sig[order(sig$adj.P.Val),]
sig_top<-sig[1:10,]

sig_up<-subset(sig,logFC>0)
sig_up_top<-subset(sig_top,logFC>0)


sig_down<-subset(sig,logFC<0)
sig_down_top<-subset(sig_top,logFC<(0))


ColorPalette<-brewer.pal(11,"RdBu")
ColorPalette<-ColorPalette[c(2:4,8:10)]

plot(Results$logFC,-log(Results$P.Value),ylim=c(0,25), xlim=c(-2.5, 2.5), pch=20,cex=0.35,main=Title,ylab="Negative Log p-value",xlab="Log Fold Change",col="grey60")
points(sig_up$logFC,-log(sig_up$P.Value),pch=20,cex=0.75,col=ColorPalette[3])
text(sig_up_top$logFC,-log(sig_up_top$P.Value),sig_up_top$hgnc_symbol,cex=0.75,col="Black",pos=1)

points(sig_down$logFC,-log(sig_down$P.Value),pch=20,cex=0.75,col=ColorPalette[4])
text(sig_down_top$logFC,-log(sig_down_top$P.Value),sig_down_top$hgnc_symbol,cex=0.75,col="Black",pos=1)
}

MakeVPlot_Lambdoid<-function(Results,Title){
sig<-subset(Results,adj.P.Val<0.05)
sig<-sig[order(sig$adj.P.Val),]
sig_top<-sig[1:10,]

sig_up<-subset(sig,logFC>0)
sig_up_top<-subset(sig_top,logFC>0)

ColorPalette<-brewer.pal(11,"RdBu")
ColorPalette<-ColorPalette[c(2:4,8:10)]

plot(Results$logFC,-log(Results$P.Value),ylim=c(0,25), xlim=c(-2.5, 2.5), pch=20,cex=0.35,main=Title,ylab="Negative Log p-value",xlab="Log Fold Change",col="grey60")
points(sig_up$logFC,-log(sig_up$P.Value),pch=20,cex=0.75,col=ColorPalette[3])
text(sig_up_top$logFC,-log(sig_up_top$P.Value),sig_up_top$hgnc_symbol,cex=0.75,col="Black",pos=1)
}

MakeVPlot(AllResults_Coronal,"Coronal")
MakeVPlot_Lambdoid(AllResults_Lambdoid,"Lambdoid")
MakeVPlot(AllResults_Metopic,"Metopic")
MakeVPlot(AllResults_Sagittal,"Sagittal")
```


## Plot DEG Directionality
Visualize DEG number and directionality across phenotypes
```{r Plot DEGs}
Sig_Coronal_Up <- Sig_Coronal %>%
  dplyr::filter(logFC>0)
Sig_Lambdoid_Up <- Sig_Lambdoid %>%
  dplyr::filter(logFC>0)
Sig_Metopic_Up <- Sig_Metopic %>%
  dplyr::filter(logFC>0)
Sig_Sagittal_Up <- Sig_Sagittal %>%
  dplyr::filter(logFC>0)

DEGNumbers_Plot <- c(4, 5, 3, 0, 27, 42, 10, 20)
Phenotype_Plot <- c("Coronal", "Coronal", "Lambdoid", "Lambdoid", "Metopic", "Metopic", "Sagittal", "Sagittal")
Direction_Plot<-c("Up", "Down", "Up", "Down", "Up", "Down", "Up", "Down")

bardf<- data.frame(Count=DEGNumbers_Plot, Phenotype=Phenotype_Plot, Direction=Direction_Plot)

bardf$Direction <- factor(bardf$Direction, levels=c("Down", "Up"))
bardf$Phenotype <- factor(bardf$Phenotype, levels=c("Coronal", "Lambdoid", "Metopic", "Sagittal"))


bardf %>% ggplot(aes(x=Phenotype, y=Count, fill=Direction)) + 
  geom_bar(stat="identity", position="dodge") + 
  theme_bw() + 
  theme(axis.title=element_text(size=16), axis.text = element_text(size=12), plot.title=element_text(size=18), legend.text=element_text(size=12), legend.title=element_text(size=16)) + 
  ggtitle("Male-DEG Direction by Craniosynostosis Phenotype") + 
  geom_text(position=position_dodge(1), aes(label=Count)) + 
  scale_fill_manual(values=c("#9be1f5", "#068eed")) + 
  ylab("Number of DEGs")

```

## Plot DEG Overlap
Visualize DEG overlap across phenotypes as an UpSet plot
```{r Upset Plot}
#Step 1: Make Vectors

Cor_Vec<- as.vector(Sig_Coronal$ensembl_gene_id)
Lamb_Vec<- as.vector(Sig_Lambdoid$ensembl_gene_id)
Met_Vec<-as.vector(Sig_Metopic$ensembl_gene_id)
Sag_Vec<- as.vector(Sig_Sagittal$ensembl_gene_id)

#Step 2: Turn vectors into a list
Upset_list<-list("Coronal"=Cor_Vec, "Lambdoid"=Lamb_Vec, "Metopic"=Met_Vec, "Sagittal"=Sag_Vec)


#Step 3: UpSet Plot
upset(fromList(Upset_list), nsets=4, order.by="freq", text.scale=2)

```

##Pathway Analysis using Fry 
Step 1: Prepare KEGG Database and remove disease-related pathways
Step 2: Map Pathways to entrez IDs for each phenotype
Step 3: Create expression matrix with entrez IDs
Step 4: Run Fry
Step 5: Pull Significant Results
```{r Fry Pathway Analysis}
#Step 1: Prepare Kegg Database
keggs <- getGeneKEGGLinks()
keggnum <- as.numeric(gsub("path:hsa", "", keggs[,2]))
kegg_dontwant <- c(4932, 4931, 4933, 4934, 4936, 4940, 4950, 4930,4949,4959,1501:1503, 1521:1524)
rmind <- keggnum %in% kegg_dontwant | keggnum > 5000
keggs <- keggs[!rmind,]
keggs$SYMBOL <- mapIds(org.Hs.eg.db, keggs[,1], "SYMBOL", "ENTREZID")
kegglst <- split(keggs$SYMBOL, keggs$PathwayID)
keggmapper <- read.table("http://rest.kegg.jp/list/pathway/hsa/", sep = "\t", quote = "\"", fill = TRUE, comment.char = "")
keggmapper[,2] <- sapply(strsplit(keggmapper[,2], " - "), function(x) gsub(" ", "_", paste(x[-length(x)], collapse = " ")))
keggmapper <- setNames(keggmapper[,2], keggmapper[,1])
names(kegglst) <- keggmapper[names(kegglst)]

#Step 2: Prepare pathway mapping to entrez ids for each phenotype
kegglst_Coronal <- lapply(kegglst, function(x) which(AllResults_Coronal[,5] %in% x))
kegglst_Coronal <- kegglst_Coronal[sapply(kegglst_Coronal, length) > 0L]

kegglst_Lambdoid <- lapply(kegglst, function(x) which(AllResults_Lambdoid[,5] %in% x))
kegglst_Lambdoid <- kegglst_Lambdoid[sapply(kegglst_Lambdoid, length) > 0L]

kegglst_Metopic <- lapply(kegglst, function(x) which(AllResults_Metopic[,5] %in% x))
kegglst_Metopic <- kegglst_Metopic[sapply(kegglst_Metopic, length) > 0L]

kegglst_Sagittal <- lapply(kegglst, function(x) which(AllResults_Sagittal[,5] %in% x))
kegglst_Sagittal <- kegglst_Sagittal[sapply(kegglst_Sagittal, length) > 0L]

#Step 3: Create expression matrix with entrez IDs 
ExprMat <- v$E
rownames(ExprMat) <- v$genes$entrezgene_id

#Step 4: Run Fry for each phenotype
fry_Coronal <- fry(ExprMat, kegglst_Coronal, mod, contrast=2)
fry_Lambdoid <- fry(ExprMat, kegglst_Lambdoid, mod, contrast=3)
fry_Metopic <- fry(ExprMat, kegglst_Metopic, mod, contrast=4)
fry_Sagittal <- fry(ExprMat, kegglst_Sagittal, mod, contrast=5)

#Step 5: Pull Fry Significant Fry Results with FDR<0.2
Fry_FDR_Coronal <- fry_Coronal %>%
  dplyr::filter(FDR<0.2) %>%
  arrange(FDR)
Fry_FDR_Coronal$ID <- rownames(Fry_FDR_Coronal)
#Printing Top 10 Coronal Pathways (FDR<0.2)
head(Fry_FDR_Coronal$ID, 10)
nrow(Fry_FDR_Coronal)

Fry_FDR_Lambdoid <- fry_Lambdoid %>%
  dplyr::filter(FDR<0.2) %>%
  arrange(FDR)
Fry_FDR_Lambdoid$ID <- rownames(Fry_FDR_Lambdoid)
#Print Top 10 Lambdoid Pathways (FDR<0.2)
head(Fry_FDR_Lambdoid$ID, 10)
nrow(Fry_FDR_Lambdoid)

Fry_FDR_Metopic <- fry_Metopic %>%
  dplyr::filter(FDR<0.2) %>%
  arrange(FDR)
Fry_FDR_Metopic$ID <- rownames(Fry_FDR_Metopic)
#Print Top 10 Metopic Pathways (FDR<0.2)
#Top 4 have FDR<0.05
head(Fry_FDR_Metopic$ID, 10)
nrow(Fry_FDR_Metopic)

Fry_FDR_Sagittal <- fry_Sagittal %>%
  dplyr::filter(FDR<0.2) %>%
  arrange(FDR)
Fry_FDR_Sagittal$ID <- rownames(Fry_FDR_Sagittal)
#Print Top 10 Sagittal Pathways (FDR<0.2)
head(Fry_FDR_Sagittal$ID, 10)
nrow(Fry_FDR_Sagittal)
```

## P-Value Distributions for Data without Quality Weights
This data was transformed using only voom before conducted differential expression analysis with limma.
```{r pval dist}
hist(AllResults_Coronal$P.Value, main="Coronal P-Values")
hist(AllResults_Lambdoid$P.Value, main="Lambdoid P-Values")
hist(AllResults_Metopic$P.Value, main="Metopic P-Values")
hist(AllResults_Sagittal$P.Value, main="Sagittal P-Values")
```

## Make Heatmap of All DEGs
```{r heatmap}
#Step 1: Create a dataframe of all DEGs
Sig_AllCorLamb <- full_join(Sig_Coronal, Sig_Lambdoid, by=c("hgnc_symbol"), suffix=c("_Coronal", "_Lambdoid"))
Sig_AllCorLambSag <- full_join(Sig_AllCorLamb, Sig_Sagittal, by=c("hgnc_symbol"), suffix=c("_", "_Sagittal"))
Sig_AllPhenotypes <- full_join(Sig_AllCorLambSag, Sig_Metopic, by=c("hgnc_symbol"), suffix=c("_Sagittal", "_Metopic")) %>%
  dplyr::select("hgnc_symbol", "logFC_Coronal", "logFC_Lambdoid", "logFC_Sagittal", "logFC_Metopic")

#Step 2: Fix Column Names
rownames(Sig_AllPhenotypes) <- Sig_AllPhenotypes$hgnc_symbol
Sig_AllPhenotypes <- Sig_AllPhenotypes[,c(-1)]
colnames(Sig_AllPhenotypes) <- c("Coronal", "Lambdoid", "Sagittal", "Metopic")

#Step 3: Export and sort by average logFC then reload
write.csv(Sig_AllPhenotypes, "/Volumes/paquette_a/slapehn/CranioRNAseq/Male_CranioRNAseq/Figures/Heatmap/SigAllPhenotypes.csv")
Sig_AllPhenotypes_Sorted <- read.csv("/Volumes/paquette_a/slapehn/CranioRNAseq/Male_CranioRNAseq/Figures/Heatmap/SigAllPhenotypes_Sorted.csv")

#Step 4: Fix rownames 
rownames(Sig_AllPhenotypes_Sorted) <- Sig_AllPhenotypes_Sorted$X
Sig_AllPhenotypes_Sorted <- Sig_AllPhenotypes_Sorted[,c(-1)]

#Step 3: Create Data Matrix
Sig_AllPhenotypesMatrix <- data.matrix(Sig_AllPhenotypes_Sorted)
Sig_AllPhenotypesMatrix_Final<-t(Sig_AllPhenotypesMatrix)

#Step 4: Make Heatmap
Heatmap_Breaks<-seq(-2.5,2.5, length.out=31)
pheatmap(Sig_AllPhenotypesMatrix_Final, 
         color=colorRampPalette(c("#0092B3", "White", "#F09062"))(31),
         #scale='column', # For Scaled Data
         breaks=Heatmap_Breaks,
         scale='none',# unscaled data
         cluster_rows =F,
         cluster_cols = F,
         na_col="grey",
         cellwidth=8, #Change tehe for size
         cellheight=8, #change these for size
         #number_color = X,
         treeheight_row = 0,
         showcolnames=T,
         fontsize_number = 0.5)
```


#Combine all DEGs across phenotypes
Combine all DEGs across phenotypes to be used for further comparisons across models (Primary, Male, Female)
```{r}
Sig_Cor_Lamb <- full_join(Sig_Coronal, Sig_Lambdoid, by=c("hgnc_symbol"), suffix=c("_Coronal", "_Lambdoid"))

Sig_Met_Sag <-full_join(Sig_Metopic, Sig_Sagittal, by=c("hgnc_symbol"), suffix=c("_Metopic", "_Sagittal"))

All_Male_DEGs<- full_join(Sig_Cor_Lamb, Sig_Met_Sag, by=c("hgnc_symbol"))

write.csv(All_Male_DEGs, "/Volumes/paquette_a/slapehn/CranioRNAseq/Male_CranioRNAseq/DEGs/All_Male_DEGs.csv")
```

